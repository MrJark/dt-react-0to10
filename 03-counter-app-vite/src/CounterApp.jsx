// Tarea: Counter App <https://gist.github.com/Klerith/e1a731cc595c00a9794a709062eae757>
// Conseguida!!

import { Fragment } from "react";
import './style.css';
import PropTypes from 'prop-types';


const CounterApp = ( { value } ) => {

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { value * 3 } </h2>

        </>
    )
};

CounterApp.propTypes = {
    value:  PropTypes.number.isRequired,
};

export default CounterApp