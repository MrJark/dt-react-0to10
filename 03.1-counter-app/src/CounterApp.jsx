// Tarea: Counter App <https://gist.github.com/Klerith/e1a731cc595c00a9794a709062eae757>
// Conseguida!!

import { useState } from "react"; // los Hooks en React empiezan con la palabra ' use '
// import './style.css'; // para hacer las pruebas ha de ser desactivado
import PropTypes from 'prop-types';


// const handleApp = () => { // si no interfiere el value del function component en la function, se coloca fuera para evitar espacio innecesario
//     // console.log(event);

// }; 

const CounterApp = ( { value } ) => {
    // const handleApp = (event) => { // si no tiene ninguna interacción con el value o dependencia con la functionComponent, es mejor tenerlo fuera para que no se renderice todas las veces que la mandes a llamar y no ocupe espacio en memoria
    //     console.log(event);
    // } 
    // const [ counter, setCounter ] = useState(0); // useState en este caso manda el valor inicial a la constante, en este caso counter y el segundo argumento que se le manda es para poder modificar el primer elemento, counter, y se sule empezar por la palabra ' set ' y aquí lo vaoms a usar para modificar el counter ya que el cauter en si no se puede modificar por ser una const
    const [ counter, setCounter ] = useState(value); // si aquí pongo el value, recibe el valor que le has asignado al padre que tenfa por defecto, en este caso el del main.jsx que es 1 
    const sumApp = () => {
        // console.log(event);
        // setCounter( counter + 1); // modificación permitida del counter
        setCounter( (t) => t +1 ); // otra manera de hacer la suma donde no hace falta saber el nombre de counter, simplemente puedes poner el nombre de lo que quieras en la función
    };
    // Mi Forma de realizar la tarea ( la mas larga pero se puede sinplificar quitando los {})
    const substractApp = () => {
        setCounter( (t) => t - 1);
    };
    // const substractApp = () => setCounter( (t) => t - 1);
    const resetApp = () => {
        setCounter( value );
    };
    // const resetApp = () => setCounter( value );
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2> {} 
            <button onClick={ sumApp /* se podría poner tb s(event) => handleApp(event) */ /*function(event) {console.log();*/ }> {/* creación de un boton con atributos donde pueden ser funciones. Aunque las buenas prácticas dicen que los elementos .js no deben estar dentro del html*/}
                +1
            </button>
            {/* Tarea: dar funcionalidad a los botones -1 y Reset. Conseguida!🎉 */}
            <button onClick={ substractApp }> -1 </button>
            <button aria-label="btn-reset" onClick={resetApp}> Reset </button>
        </>
    )
};

CounterApp.propTypes = { // la ' p ' minúscula !OJO¡
    value:  PropTypes.number.isRequired,
};

export default CounterApp