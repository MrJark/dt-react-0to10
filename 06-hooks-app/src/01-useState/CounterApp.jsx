import { useState } from 'react';


export const CounterApp = () => {

    // const [counter, setCounter] = useState(1)

    // Tarea: hacer que funcionen los distintos counters a la vez ( conseguido! GuÃ­a -> https://react.dev/reference/react/useState#examples-objects Apartado: Examples of objects and arrays in state )
    const [counter, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
    });

    const { counter1, counter2, counter3 } = counter; // desestructuraciÃ³n del objeto counter pero no en la misma linea

    


  return (
    <>
        <h1>Counter1: {counter1}</h1>
        <h1>Counter2: {counter2}</h1>
        <h1>Counter3: {counter3}</h1>

        <hr />
        

        <button onClick={
            () => {
                    setCounter({
                        ...counter, // desestruturo y cojo lo que me interesa
                        counter1: counter1+1, // le sobreescribo lo que yo quiera
                    })
                }
        }>+1</button>

        <button onClick={
            () => {
                    setCounter({
                        ...counter, 
                        counter1: counter1 - 1,
                    })
                }
        }>-1</button>

      

        <button className='btn' onClick={() => setCounter (counter1 + 1)}>+1</button>
        <button className='btn' onClick={() => setCounter (counter1 - 1)}>-1</button>

    </>
  )
}
