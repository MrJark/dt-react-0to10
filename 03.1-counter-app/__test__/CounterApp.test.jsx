import { render, screen } from "@testing-library/react";
// import App from "../src/FirstApp";
import CounterApp from "../src/CounterApp";


describe('pruebas a CounterApp', () => {

    const initialValue = 1;

    test('Tarea 1: hacer match con el snapshot (cnoseguido)', () => {
        const { container } = render(
            <CounterApp
                value={initialValue}
            />
        );
        expect( container ).toMatchSnapshot();
    });

    test('Tarea 2: debe mostrar el valor inicial de 1(no conseguido)', () => {

        render(
            <CounterApp
                value={initialValue}
            />
        );
        expect( screen.getByText(initialValue)).toBeTruthy();

    });

});