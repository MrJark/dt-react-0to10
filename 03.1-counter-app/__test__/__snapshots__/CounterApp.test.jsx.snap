// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas a CounterApp Tarea 1: hacer match con el snapshot (cnoseguido) 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    1
     
  </h2>
   
  <button>
     
    +1
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
